server/services/return-of-service.js

```diff
@@ -1,6 +1,111 @@
 /* eslint-disable camelcase */
 const { dbClient, collections } = require('../db');
 const { rosError } = require('../constants');
+const logger = require('../logger.js');
+
+const getParticipantStatuses = async (participantId) => {
+  const statuses = await dbClient.db[collections.PARTICIPANTS_STATUS].find({
+    participant_id: participantId,
+    status: 'hired',
+    current: true,
+  });
+  if (statuses.length === 0) {
+    throw new Error(rosError.participantNotHired);
+  }
+  return statuses;
+};
+
+const getRosParticipant = async (participantId) => {
+  const rosParticipants = await dbClient.db[collections.ROS_STATUS].find({
+    participant_id: participantId,
+    is_current: true,
+  });
+  if (rosParticipants.length === 0) {
+    throw new Error(rosError.participantNotFound);
+  }
+  return rosParticipants[0];
+};
+
+const getDbSiteId = async (site) => {
+  const sites = await dbClient.db[collections.EMPLOYER_SITES].find({
+    'body.siteId': site,
+  });
+  if (sites.length === 0) {
+    throw new Error(rosError.noSiteAttached);
+  }
+  return sites[0].id;
+};
+
+const validateRosUpdateBody = (rosParticipant, updatedSite, updatedDate, updatedStartDate) => {
+  if (!rosParticipant.data) {
+    throw new Error(rosError.fieldNotFound);
+  }
+  if (!updatedSite && !updatedDate && !updatedStartDate) {
+    throw new Error(rosError.noFieldsToUpdate);
+  }
+  if (updatedSite && !rosParticipant.site_id) {
+    throw new Error(rosError.noSiteAttached);
+  }
+  if (updatedDate && !rosParticipant.data.date) {
+    throw new Error(rosError.noDate);
+  }
+  if (updatedStartDate && !rosParticipant.data.startDate) {
+    throw new Error(rosError.noStartDate);
+  }
+};
+
+const formatRosData = async (
+  participantId,
+  data,
+  siteId,
+  newSiteId,
+  user,
+  status,
+  assignNewSite,
+  isMohUpdate
+) => {
+  const statuses = await getParticipantStatuses(participantId);
+  let rosParticipant;
+  if (isMohUpdate || assignNewSite) {
+    rosParticipant = await getRosParticipant(participantId);
+  }
+
+  let site_id;
+  if (isMohUpdate && data.site) {
+    site_id = await getDbSiteId(data.site);
+  }
+  if (!isMohUpdate) {
+    site_id = siteId;
```

Comment:
Is this db ID vs site id? This could use a rename or a comment to better explain what's going on here